pandas/core/base.py

```diff
@@ -881,10 +881,8 @@ def _map_values(self, mapper, na_action=None):
         """
         arr = extract_array(self, extract_numpy=True, extract_range=True)
 
-        if is_extension_array_dtype(arr.dtype):
-            # Item "IndexOpsMixin" of "Union[IndexOpsMixin, ExtensionArray,
-            # ndarray[Any, Any]]" has no attribute "map"
-            return arr.map(mapper, na_action=na_action)  # type: ignore[union-attr]
+        if isinstance(arr, ExtensionArray):
```

Comment:
`DatetimeArray`/`DatetimeIndex` etc. have numpy dtypes, so using `is_extension_array_dtype`  isn't enough, we need to check for `ExtensionArray` to reach the `.map` method of datetime-likes.